implementations/rust/kex/src/x3dh.rs

```diff
@@ -179,20 +176,19 @@ enum InitiatorState {
 
 /// The responder of X3DH receives a prekey bundle and computes the shared secret
 /// to communicate the first message to the initiator
-#[derive(Clone)]
 pub struct X3dhInitiator {
-    ephemeral_identity_key: SecretKeyContext,
+    ephemeral_identity_key: Box<dyn Secret>,
     prekey_bundle: Option<PreKeyBundle>,
     state: InitiatorState,
     vault: Arc<Mutex<dyn DynVault + Send>>,
     completed_key_exchange: Option<CompletedKeyExchange>,
-    identity_key: Option<SecretKeyContext>,
+    identity_key: Option<Arc<Box<dyn Secret>>>,
 }
 
 impl X3dhInitiator {
-    fn new(v: Arc<Mutex<dyn DynVault + Send>>, identity_key: Option<SecretKeyContext>) -> Self {
+    fn new(v: Arc<Mutex<dyn DynVault + Send>>, identity_key: Option<Arc<Box<dyn Secret>>>) -> Self {
         Self {
-            ephemeral_identity_key: SecretKeyContext::Memory(0),
+            ephemeral_identity_key: Box::new(DefaultVaultSecret::default()), // FIXME: Should be Option<Secret>
```

Comment:
Why isn't it an option?

Reply:
Needs some changes in the places of usage, I'll do it in further PRs